tests/lib/rules/no-implicit-coercion.js

```diff
@@ -46,6 +51,15 @@ ruleTester.run("no-implicit-coercion", rule, {
         {code: "+foo.bar", errors: [{message: "use `Number(foo.bar)` instead.", type: "UnaryExpression"}]},
         {code: "1*foo", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
         {code: "foo*1", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
+        {code: "1234*foo*1", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
+        {code: "1234*1*foo", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
+        {code: "1234*bar*1*foo", errors: [
+            {message: "use `Number(bar)` instead.", type: "BinaryExpression"},
+            {message: "use `Number(foo)` instead.", type: "BinaryExpression"}
+        ]},
+        {code: "1234*1*foo*bar", errors: [{message: "use `Number(foo)` instead.", type: "BinaryExpression"}]},
```

Comment:
Not sure if this one not being reported is a bug or not since `bar` is not in direct contact with `1`. If we are okay with this one I should add one more valid test case for `1234 * Number(bar) * 1 * Number(foo)`.